src/msg/async/AsyncMessenger.cc

```diff
@@ -167,7 +167,7 @@ int Processor::rebind(const set<int>& avoid_ports)
   ldout(msgr->cct, 10) << __func__ << " new nonce " << nonce << " and inst " << msgr->my_inst << dendl;
 
   ldout(msgr->cct, 10) << __func__ << " will try " << addr << " and avoid ports " << new_avoid << dendl;
-  return bind(addr, new_avoid);
+  return bind(msgr->my_inst.addr, new_avoid);
```

Comment:
@wjwithagen i discussed this with @yuyuyu101 over IRC just now. it's not a bug of clang. we should pass the `addr` with the new nonce, and the current code fails to do so. i am posting a new PR replacing yours so it can be reviewed by @yuyuyu101 sooner. also it's actually a bug in ceph, i am filing a ticket on tracker at http://tracker.ceph.com/issues/17807